scripts/blocks-utils.js

```diff
@@ -116,9 +130,30 @@ function observe(elementToObserve, callback, ...args) {
  * } else {
  *  console.log(data);
  * }
+ * }); // GET request
+ *
+ * fetchData('https://example.com/data', (error, data) => {
+ *   if (error) {
+ *     console.error('Error fetching data:', error);
+ *   } else {
+ *     console.log('Data fetched successfully:', data);
+ *   }
+ * }, 'getdata'); // POST request with API name
  */
-function fetchData(url, callback) {
-  fetch(url)
+function fetchData(url, callback, apiName = null) {
```

Comment:
I also thought of implementing the method like 
`fetchData(url, apiname, callback)`

but this would impact all the uses of fetchData we have in the code.
If we all agree on one signature of the method, I can update the rest of the code as well if needed.

Reply:
IMO go with `fetchData(url, apiname, callback)` as callback will be inline method so this signature is more clean to understand, and `url` and `apiName` is related then on the result `callback` will execute.
And lets create a new method so existing will not break, once every block boarded we will remove previous one.

Reply:
@ravverma  but `apiname` is an option param as well, so it would make sense to keep it in the end.
I am open to new method if all agree .

Reply:
for making it more generic, we can consider adding following changes:
1. add a new method postFormData (url: string, formData: FormData, callback: function, options?: RequestOptions) , this will make sure any form body can be posted, any additional param like headers can be passed which can be optional
2. fetchData can be there to make get requests with params like fetchData(url: string, callback: function, options?: RequestOptions), apiname has to be removed as it not generic enough to be added in utils
3. If possible we should move fetch and get related methods in a new file.

Reply:
It seems general opinion is having new method. Let me add it.
IMO we are keeping  everything related to utils in block-utils , so lets keep it in the same file as of now.

Reply:
@vivesing  if we introduce formData , then user has to create formData at their end. While I agree it that providing formData is something user more power, but decreases the ease for user.
Let me create two separate method and calling the main function as it.

Reply:
for post requests users will have to pass the body/form-data as per need, there are few APIs where form-data have additional and new values apart from apiname, plus consumers having control of the form data gives more flexibility in the long run. Each block should be handling the body sent and response parsing in its block code.

Reply:
@vivesing yes for example where pagination is in play. 
I have provided both method to provide more control to the user